gamification/process/update.py

```diff
@@ -0,0 +1,142 @@
+import logging
+
+from data.models import (
+    MergeRequest,
+    Issue,
+)
+from gamification.process.activity_points import (
+    get_activity_with_points
+)
+from gamification.models import Participant
+
+
+def get_mr_objects():
+    """
+    Get mrs objects saved in the database.
+
+    :return: a queryset of mr objects.
+    """
+    mrs = MergeRequest.objects.all()
+    return mrs
+
+
+def update_participants_data_with_mr(mr):
+    """
+    Update particiapnts data based on mr.
+
+    This method updates the total score earned by the
+    participant based on the activities performed,
+    and then update the current level based on the
+    total score.
+
+    This method first check if the mr is merged or not,
+    if it's merged, then it's get a activity and points based
+    on the labels on mr and update the participant data who
+    opened this mr.
+
+    Further it gets all the issues that mr is closing then get the
+    points and activity based on the labels on the issue and update
+    the participant data who opened that issue.
+    """
+    logger = logging.getLogger(__name__)
+
+    # Get the participant who opened this mr
+    mr_author = Participant.objects.get(username=mr.author)
+    mr_performed_at = mr.created_at
+    mr_updated_at = mr.updated_at
+
+    if mr.state == 'merged':
+        mr_labels = mr.labels.values('name')
+
+        # Get activity and points based on labels on the mr
+        mr_points, mr_activity_string = get_activity_with_points(
+            'merge_request', mr_labels)
+
+        # Update participant data
+        mr_author.add_points(mr_points,
+                             mr_activity_string,
+                             mr_performed_at,
+                             mr_updated_at,
+                             )
+        mr_author.save()
+
+        # Get all the issues this mr is closing
+        try:
+            issues = mr.get_closes_issues_object()
+        except Exception as ex:
+            logger.error(ex)
+            return
+        for issue in issues:
+            issue_labels = issue.labels.values('name')
+
+            # Get the participant who opened the issue
+            issue_author = Participant.objects.get(username=issue.author)
+            issue_performed_at = issue.created_at
+            issue_updated_at = issue.updated_at
+
+            # Get activity and points based on the labels on the issue
+            issue_points, issue_activity_string = get_activity_with_points(
+                'issue', issue_labels)
+
+            # Update participant data who opened the issue
+            issue_author.add_points(issue_points,
+                                    issue_activity_string,
+                                    issue_performed_at,
+                                    issue_updated_at,
+                                    )
+            issue_author.save()
```

Comment:
if issue_author is mr_author, do not save , as it will be saved at the end of the function, and we dont want to do a lot of db updates unnecessarily